modules/p2/ding_list/include/menu_callbacks.inc

```diff
@@ -172,13 +172,69 @@ function ding_list_private_share_message_callback($list_id) {
 
   $commands = array();
 
+  drupal_add_js(drupal_get_path('module', 'ding_list') . '/js/ding-list-private-sharer.js');
+
+  $query = array();
+
+  if ($_GET['community'] !== FALSE) {
+    $query['community'] = $_GET['community'];
+  }
```

Comment:
Why is `$community` a query parameter and not part of the menu path? Other parts of the code like `window.sharer` seem to require it anyway.